src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -27,18 +27,22 @@ private static String trim(String imageId) {
     /**
      * Gets {@link Fingerprint} for a given docker image.
      */
-    public static @CheckForNull Fingerprint of(String imageId) throws IOException {
+    public static @CheckForNull Fingerprint of(@Nonnull String imageId) throws IOException {
         return Jenkins.getInstance().getFingerprintMap().get(trim(imageId));
     }
 
-    private static @Nonnull Fingerprint make(Run<?,?> run, String imageId) throws IOException {
+    private static @Nonnull Fingerprint make(@Nonnull Run<?,?> run, @Nonnull String imageId) throws IOException {
         return Jenkins.getInstance().getFingerprintMap().getOrCreate(run, "<docker-image>", trim(imageId));
     }
 
     /**
      * Adds a new {@link ContainerRecord} for the specified image, creating necessary intermediate objects as it goes.
      */
-    public static void addRunFacet(String imageId, ContainerRecord record, Run<?,?> run) throws IOException {
+    public static void addRunFacet(@Nonnull ContainerRecord record, @Nonnull Run<?,?> run) throws IOException {
+        String imageId = record.getImageId();
+        if (imageId == null) {
```

Comment:
`imageId` is expected to be non-null at all times except _immediately and temporarily_ after deserialization, so I think `@Nonnull` is appropriate.